src/structs/model.rs

```diff
@@ -501,12 +528,12 @@ impl ModelPerColor {
 
     fn get_residual_threshold_counts_mut(
         &mut self,
-        ptcc8: &ProbabilityTablesCoefficientContext,
+        best_prior_abs: u32,
         min_threshold: i32,
         length: i32,
     ) -> &mut [Branch; RESIDUAL_THRESHOLD_COUNTS_D3] {
         return &mut self.residual_threshold_counts[cmp::min(
-            (ptcc8.best_prior.abs() >> min_threshold) as usize,
+            (best_prior_abs >> min_threshold) as usize,
```

Comment:
I'm not totally sure, but I think there is a hypothetical scenario where it can go wrong here: If `best_prior` is something like 0x10001, you will have in initial Lepton with intermediate reduction to 16-bit value and maximum possible `min_threshold=3` index = 0, while in the code index = 0x10 = 16, definitely different. See [the code here](https://github.com/dropbox/lepton/blob/f34c7f42d2e0d688239cb880648dc152713c4eae/src/vp8/model/model.hh#L1079)

Reply:
so it should be (best_prior_abs & 0xffff) >> min_threshold

Reply:
Ok added the change together with a comment